src/main/java/uk/gov/hmcts/reform/jobscheduler/services/jobs/GetterFromScheduler.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.hmcts.reform.jobscheduler.services.jobs;
+
+import org.quartz.SchedulerException;
+import uk.gov.hmcts.reform.jobscheduler.services.jobs.exceptions.JobException;
+
+class GetterFromScheduler {
+
+    @FunctionalInterface
+    interface Getter<T, R> {
+        R apply(T t) throws SchedulerException;
+    }
+
+    private GetterFromScheduler() {
+    }
+
+    static <T, R> R getFromScheduler(Getter<T, R> getter, T arg, String message) {
+        try {
+            return getter.apply(arg);
+        } catch (SchedulerException exc) {
+            throw new JobException(message, exc);
+        }
+    }
+
+    static <T, R> R getFromScheduler(Getter<T, R> getter, T arg) {
+        return getFromScheduler(getter, arg, "Error while getting information from scheduler");
+    }
+}
```

Comment:
can this be simpler?

Reply:
In service.getAll I am facing two nesting try/catch blocks with same exception.  It is simpler:
- get rid of nesting try catch blocks
- get rid of root try catch block too
- it is a general java8 lambda function feature with custom apply function which defines desired exception throw

Reply:
Each of you has a point. How about making something that would result in invocations looking like this:

```
   Set<JobKey> jobKeys = tryExecuting(
        () -> scheduler.getJobKeys(someArgument),
        thrownException -> throw new JobException("message", thrownException)
    );
```
The first argument would be something like your `Getter` interface (only T is unnecessary) and the second one is an error-handling `Consumer<? extends RuntimeException>`.

The benefit is that, in my opinion, the invocation would reveal more information about what's happening. Or maybe some other idea?

Reply:
suits for generalised try/catch, which probably is relevant for entire service class which has different scheduler actions to execute. path to glorifying union after seeing one idea with an intent to refactor all cases